docs/guide/custom-ui.md

```diff
@@ -198,6 +198,49 @@ The `AposWidget` component has **nothing to do with a typical site visitor exper
 Before you override an editor modal, consider [adding a custom schema field type](/guide/custom-schema-field-types.md) instead.
 :::
 
+## Adding custom context menu items
+
+Starting from Apostrophe v3.18.0 we can add custom context menus (edit mode) from within any module, targeting any Vue component that implements `AposModal`. The menu registration should happen in the initialization phase.
+
+Here is an example of how to add custom context menu labeled "My Menu".
+
+![A custom context menu 'My Menu' in the Piece Editor Modal](/images/ui-custom-context-menu.png)
+
+<AposCodeBlock>
+```js
+module.exports = {
+  extend: '@apostrophecms/piece-type',
+  options: {
+    label: 'Article',
+    pluralLabel: 'Articles'
+  },
+  init(self) {
+    self.apos.doc.addContextOperation(self.name, {
+      context: 'update',
+      action: 'myUniqueAction',
+      label: 'My Menu',
+      modal: 'MyModalComponent'
+    });
+  }
+}
+```
+  <template v-slot:caption>
+    modules/article/index.js
+  </template>
+</AposCodeBlock>
+
+::: warning
+Do not use core actions as your `action` property value - this would lead to unpredictable results and generally broken UI. You may consult what the core actions are in the [AposDocContextMenu component](https://github.com/apostrophecms/apostrophe/blob/main/modules/%40apostrophecms/doc-type/ui/apos/components/AposDocContextMenu.vue) (computed property `menu`).
+:::
+
+::: note
+* The current API supports only `context: "update"`.
```

Comment:
Specify what that does (it's only available for previously saved documents).

Reply:
I think the current behavior when `context` is not `update` is buggy (it should just allow the operation all the time but instead it forbids it all the time). So I like your language here ("supports only"), which doesn't make explicit the current buggy behavior if context is not specified.